Please carefully check the path of the folder I will take here. I will demonstrated the data reading part here.

Download the audio data, mount the drive. Similar process 

In [ ]:
# file directory check
# On cats and dogs dataset, under training folder, you will see there are two folders called cats and dogs. we will take those
import os
for dirname, _, filenames in os.walk('/content/drive/MyDrive/Programs/cats_dogs/train'):
    for filename in filenames:
        os.path.join(dirname, filename)

In [ ]:
# import basic packages
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline
import librosa, librosa.display
from tqdm import tqdm

In [ ]:
# check the sample rate
# load a sample data
x, sr = librosa.load(r'/content/drive/MyDrive/Programs/cats_dogs/test/cats/cat_110.wav')
print(sr)

22050


But we are not sure, whether each sample has 22050 amount of data length. Therefore, we can take a definite number of sample points.

In this dataset, one data has 20765 number of sample points. 

Therefore, to make sure all the lengths are similar we will do the following to
create our dataset.

In [ ]:
# on the datadirectory also we will give that previously mentioned path
DATADIR = '/content/drive/MyDrive/Programs/cats_dogs/train'
CATEGORIES = ['cat','dog']

In [ ]:
# perfrom some processing to store the data
training_data=[]
def create_training_data():
    for category in CATEGORIES:
        path=os.path.join(DATADIR, category)
        class_num=CATEGORIES.index(category)
        for audio in os.listdir(path):
            audio_array, sr_array=librosa.load(os.path.join(path,audio))
            audio_array = audio_array[0:20765]
            training_data.append([audio_array,class_num])
create_training_data()  

If we are sure that each dataset has at least samples till sr amount, then on the previous block of code 

we can wtite 


```
# audio_array = audio_array[0:sr]
```



In [ ]:
# check the length of training data and store it for further usage
print(len(training_data))
lenofaudio = len(training_data)

210


In [ ]:
# split x and y
X=[]
y=[]

for categories, label in training_data:
    X.append(categories)
    y.append(label)


In [ ]:
X = np.vstack(X)
X = np.array(X).reshape(lenofaudio,-1)
X.shape

(210, 20765)

In [ ]:
# y should be in array form as well
y=np.array(y)
# check the shape
y.shape

(210,)

# **Classifier Sample**

In [ ]:
# Now we are ready with our dependent and independent features, now its time for data modeling.
# applying train_test_split on our data
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X,y)

In [ ]:
from sklearn.svm import SVC
svc = SVC(kernel='linear',gamma='auto')
svc.fit(X_train, y_train)

SVC(C=1.0, break_ties=False, cache_size=200, class_weight=None, coef0=0.0,
    decision_function_shape='ovr', degree=3, gamma='auto', kernel='linear',
    max_iter=-1, probability=False, random_state=None, shrinking=True,
    tol=0.001, verbose=False)